credentials.py

```diff
@@ -0,0 +1,2 @@
+CSRF_TOKEN='sbsJZfH6UGSx5Y5k6A=='
+COOKIE='aad3b0a4-86d2-416a-9667-c0709ef25e3e'
```

Comment:
Don't make this credentials visible make sure they are ignored by git by adding them in .gitignore, and I suppose cookie is a big key value string you can grab that from drv console

Reply:
Ooh is this not the right key value string for cookie? I checked on chrome network tab. Is that not right? Thanks for your help. I don't want the files to print I want the files to write to local drive so I can push to github after.

Reply:
Also for this do I add the credentials underneath credentials.py in the .gitignore file?

Reply:
> Ooh is this not the right key value string for cookie? 
cookie string itself is key value internally, I suggest  go to network tab, hit refresh and on `dashboard` url grab the cookies from request headers, it will be some long string of key value

 > I don't want the files to print I want the files to write to local drive so I can push to github after

This will happen if credentials are set no need to change any code.

Reply:
> Also for this do I add the credentials underneath credentials.py in the .gitignore file?

I guess you uploaded it on github by drag & drop? you shouldn't do this, instead this should be in your local only. and anyways `.gitgnore` have `credentials.py` file so it would be ignored by git(won't get uploaded to github)

Reply:
You said to change the base directory, but what should I be changing it to? I like that it is hackerrank_submissions

Reply:
Ok 

> > Ooh is this not the right key value string for cookie?
> 
> cookie string itself is key value internally, I suggest go to network tab, hit refresh and on `dashboard` url grab the cookies from request headers, it will be some long string of key value
> 
> > I don't want the files to print I want the files to write to local drive so I can push to github after
> 
> This will happen if credentials are set no need to change any code.

Okay! For the cookies I go to any hackerrank url and then I click inspect and then network tab. Then I search for CSRF and cookie?

Reply:
correct

and BTW git/github is smart even if you delete `credentials.py` file it can be viewed by surfing tree at previous commit :smile:

Reply:
Okay thank you. How should I delete it then? Not that they are the right credentials haha but still good to know

Reply:
haha :smile: yaa true, I haven't ran into such issue. but this might be [helpful](https://stackoverflow.com/a/8085853/9485283)

Reply:
I think I am an idiot haha because the files still aren't saving locally. I added credentials.py to my local drive and its in .gitignore. I grabbed x-CSRF from the network tab and cookie from the appearances tab in Chrome dev tools. Is that right?

Reply:
The x-CSRF seems consistent across pages but isn't working

Reply:
Where are you saving the file?

Reply:
I did BASE_DIR: str = "../hackerrank_submissions"

Reply:
and how are you running the script?
python scraper.py?

Reply:
or python /home/**/**/scaper.py?

Reply:
python scraper.py

Reply:
hmm, what are the logs? can you share the logs, of the console?

Reply:
The logs.txt file is blank

Reply:
naa not that, the console output

Reply:
sql Domain: aggregation                          revising-aggregations-sum
sql Domain: aggregation         revising-aggregations-the-average-function
sql Domain: aggregation                                 average-population
sql Domain: aggregation                                   japan-population
sql Domain: aggregation                      population-density-difference
sql Domain: aggregation                                        the-blunder
sql Domain: aggregation                              earnings-of-employees
sql Domain: aggregation                      weather-observation-station-2
sql Domain: aggregation                     weather-observation-station-13
sql Domain: aggregation                     weather-observation-station-14
sql Domain: aggregation                     weather-observation-station-15
sql Domain: aggregation                     weather-observation-station-16
sql Domain: aggregation                     weather-observation-station-17
sql Domain: aggregation                     weather-observation-station-18
sql Domain: aggregation                     weather-observation-station-19
sql Domain: aggregation                     weather-observation-station-20
sql Domain: join                                          asian-population
sql Domain: join                                            african-cities
sql Domain: join                      average-population-of-each-continent
sql Domain: join                                                the-report
sql Domain: join                                                full-score
sql Domain: join                                    harry-potter-and-wands
sql Domain: join                                                challenges
sql Domain: join                                       contest-leaderboard

Reply:
That's not the full one but thats a sample

Reply:
hmm, does it says permission denied on the direcotry somwhere in logs?

Reply:
Nope these are the first 4 lines: 
sql Domain: select                               revising-the-select-query
sql Domain: select                             revising-the-select-query-2
sql Domain: select                                          select-all-sql
sql Domain: select                                            select-by-id

Reply:
the directory will be created above the scraper script, if you run python scraper.py

if you run python interpreter from other directory then hackerrank_submissions will be created on directory above that..

Reply:
Full output: 
(base) Marissas-MBP-2:hackerrank-scrapper marissaposner$ git pull origin master
From https://github.com/marissaposner/hackerrank-scrapper
 * branch            master     -> FETCH_HEAD
Already up to date.
(base) Marissas-MBP-2:hackerrank-scrapper marissaposner$ python scraper.py
sql Domain: select                               revising-the-select-query
sql Domain: select                             revising-the-select-query-2
sql Domain: select                                          select-all-sql
sql Domain: select                                            select-by-id
sql Domain: select                              japanese-cities-attributes
sql Domain: select                                    japanese-cities-name
sql Domain: select                           weather-observation-station-1

Reply:
> Nope these are the first 4 lines:

I don't know why this ghost behavior is happening :tired_face: Everything should work out of the box, only cookies and csrf need to be setuped

Reply:
The files should download with the extensions specified in util.py right?

Reply:
Maybe my cookies/CSRF are wrong

Reply:
did you checked the OP [blog](https://medium.com/hackernoon/scrapping-your-hackerrank-com-submissions-with-python-d409bdf516bb)?

Reply:
> The files should download with the extensions specified in util.py right?

yes, even if not, it would be downloaded anyways

Reply:
Yes my one question is that the article specifies to grab the keys from any XHR request. So I click record on the tab and then click around to submissions and to view my submission. Is that an XHR request?

Reply:
Is the cookie the hackerrank_mixpanel_token?

Reply:
> Yes my one question is that the article specifies to grab the keys from any XHR request. So I click record on the tab and then click around to submissions and to view my submission. Is that an XHR request?

yes

> Is the cookie the hackerrank_mixpanel_token?
nope, its 
`cookie: h_r=internal-search; .... ; `

`x-csrf-token: l8VFAJ0+....==`

so in credentials.py -> 
COOKIE = "copy paste exact here"
CSRF_TOKEN="copy paste exact here without spaces"

Reply:
Okay but should the cookie say: COOKIE='hackerrank=aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e'
and csrf token should say: 
CSRF_TOKEN='AtlAyYo5wXkuAX2CsDzTFWhiIebXLTlouEhoY4uXXXXXaBq3OEB2xJx2SPfX4VDms7XXXwxeG5Xabg3Q=='

Reply:
and the cookie should be the 

> > Yes my one question is that the article specifies to grab the keys from any XHR request. So I click record on the tab and then click around to submissions and to view my submission. Is that an XHR request?
> 
> yes
> 
> > Is the cookie the hackerrank_mixpanel_token?
> > nope, its
> 
> `cookie: h_r=internal-search; .... ; `
and the cookie should be the h_r? Should it say 'hackerrank=...' or 'h_r=...' or neither just COOKIE='aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e'

Reply:
different people can have different cookie content. but in the end it's key value
like 
`cookie: KEY1= VALUE1; KEY2=VALUE2 ......`

so it's okay if you have
`cookie: hackerrank=aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e` 

under your request/response header of chrome dev tool

Reply:
> Is the cookie the hackerrank_mixpanel_token?

yes it should contain this key

Reply:
> different people can have different cookie content. but in the end it's key value
> like
> `cookie: KEY1= VALUE1; KEY2=VALUE2 ......`
I have 2 hackerrank_mixpanel_token under my cookies tab. One for .hrcdn.net and one for .hackerrank.com Which one is the right one?

Reply:
Neither of them seem to be working but I think I'm confused about what goes in the '' is it Should it say COOKIE='hackerrank=...' or COOKIE='h_r=...' or neither just COOKIE='aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e'

Reply:
>I have 2 hackerrank_mixpanel_token under my cookies tab. One for .hrcdn.net and one for .hackerrank.com Which one is the right one?

are they both same? If not use hackerrank.com one

Reply:
It's not the metrics_user_identifier cookie right?

Reply:
I nothing works, trying logging out and logging back in, and grab those again.

Reply:
> It's not the metrics_user_identifier cookie right?

best think to do rather not confusing, open dev tool > Application > Cookies> hackkerank.com > you'll see your cookies here, verify what you copied is this one

x-csrf-token isn't and issue it's short and simple

Reply:
> > It's not the metrics_user_identifier cookie right?
> 
> best think to do rather not confusing, open dev tool > Application > Cookies> hackkerank.com > you'll see your cookies here, verify what you copied is this one
> 
> x-csrf-token isn't and issue it's short and simple

That's the COOKIE I have been using, but do I need to write COOKIE='hackerrank_mixpanel_token=....MYCOOKIE' or not include the 'hackerrank_mixpanel_token' ?

Reply:
> COOKIE='hackerrank_mixpanel_token=....MYCOOKIE' or not include the 'hackerrank_mixpanel_token' ?

yes!! btw this was for just verifying(coz this table isn't friendly for copy pasting individual key value) copy the same thing from request header of any xhr request :)

Reply:
I logged out and back in so I got a new x-csrf token but it looks different than the format you wrote: 
CSRF_TOKEN='+XXblLi5wShYuYI+axBB+tte10UjxGKnx9XXdMn4IabtuzyX8Xf9gV+AuXXgPS4JhV+a3uXtvOk0pIRYsXX5rw=='
Should the +s be included?

Reply:
everything should be copied, we don't know what hackerrank servers expects, so it should look like the python script is ditto hackerrank website making request :p

Reply:
> I logged out and back in so I got a new x-csrf token but it looks different than the format you wrote:
> CSRF_TOKEN='+XXblLi5wShYuYI+axBB+tte10UjxGKnx9XXdMn4IabtuzyX8Xf9gV+AuXXgPS4JhV+a3uXtvOk0pIRYsXX5rw=='
> Should the +s be included?

not a problem it's okay

Reply:
I don't think the issue is the tokens then because nothing is downloading

Reply:
This is what I have directly (except I scrubbed out some letters) in my credentials.py
CSRF_TOKEN='+XXblLi5wShYuYI+axBB+tte10UjxGKnx9XXdMn4IabtuzyX8Xf9gV+AuXXgPS4JhV+a3uXtvOk0pIRYsXX5rw==
COOKIE='aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e'

Reply:
<img width="768" alt="Screen Shot 2019-10-20 at 3 00 42 PM" src="https://user-images.githubusercontent.com/46746064/67164615-69394080-f34a-11e9-9d0d-7316691ec6a0.png">
This is my file hierarchy

Reply:
> I don't think the issue is the tokens then because nothing is downloading

hmm I don't know where the problem persists now! I'll suggest doing it fresh, clone the repo, add credentials.py file and run the scraper

Reply:
Question: Do I have to gitpush any of the changes I made with credentials.py on my local? No right because I want it to stay local?

Reply:
> 'aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e' 

cookie would be key value? verify under Application tab

Reply:
> > I don't think the issue is the tokens then because nothing is downloading
> 
> hmm I don't know where the problem persists now! I'll suggest doing it fresh, clone the repo, add credentials.py file and run the scraper

Yeah I think I might do that -_-

Reply:
> Question: Do I have to gitpush any of the changes I made with credentials.py on my local? No right because I want it to stay local?

you don't have to push anything, it's just if you want to share globally. secondly credentials.py is in .gitignore so that won't be tracked(won't be pushed to git check git status that file won't appear )

Reply:
> > 'aad3b0a4-XXXX-416a-XXXX-c0709ef25e3e'
> 
> cookie would be key value? verify under Application tab

Under application tab I chose cookie that was = hackerrank_mixpanel_token

Reply:
yes that's what you want!